### PR TITLE
Allow "kickout" on every shape, not just frames, and add drag-and-drop example

### PR DESCRIPTION
**Pull request by TodePond**
> Stacked on top of https://github.com/tldraw/tldraw/pull/3405

This PR makes `kickoutOccludedShapes` shape-agnostic. It now runs on any shape that opts-in to the drag and drop manager by having an `onDragShapesOut` method. This is more of a bug fix than a feature.

This PR also adds an example for drag and drop.

https://github.com/tldraw/tldraw/assets/15892272/15b66be4-75bd-4c30-a594-340415bc5896

### Change Type

- [ ] `sdk` — Changes the tldraw SDK
- [ ] `dotcom` — Changes the tldraw.com web app
- [ ] `docs` — Changes to the documentation, examples, or templates.
- [ ] `vs code` — Changes to the vscode plugin
- [ ] `internal` — Does not affect user-facing stuff

- [ ] `bugfix` — Bug fix
- [ ] `feature` — New feature
- [ ] `improvement` — Improving existing features
- [ ] `chore` — Updating dependencies, other boring stuff
- [ ] `galaxy brain` — Architectural changes
- [ ] `tests` — Changes to any test code
- [ ] `tools` — Changes to infrastructure, CI, internal scripts, debugging tools, etc.
- [ ] `dunno` — I don't know

### Test Plan

1. Add a step-by-step description of how to test your PR here.
2.

- [ ] Unit Tests
- [ ] End to end tests

### Release Notes

- Add a brief release note for your PR here.

### CODE REVIEW

File: packages/tldraw/src/lib/tools/SelectTool/selectHelpers.ts [hunk 1]
```diff
@@ -2,24 +2,29 @@ import {
...
 		.map((shape) => {
 			const parent = editor.getShape(shape.parentId)
 			if (!parent) return shape
 			return parent
 		})
-		.filter((shape) => shape.type === 'frame')
+		.filter((parent) => {
+			const shapeUtil = editor.getShapeUtil(parent)
+			return shapeUtil.onDragShapesOut
+		})
```
Comment (mimecuvalo) [inline packages/tldraw/src/lib/tools/SelectTool/selectHelpers.ts:23]: small thing, but we should bail out here early if there are no affectedParents

**Comment by TodePond** [comment]
@steveruizok did some work on this and we went through some edge cases together. I've had a look through, and I've checked the nastiest cases I know of (though tests should catch those anyway). Happy to land this and QA tomorrow if others are
